tasks/monitoring.yml

```diff
@@ -0,0 +1,58 @@
+- name: Centralize VPN Logs
+  watch_logs:
+    name: openvpn
+    file_paths:
+      - "{{ openvpn_log }}"
+
+- name: Create directory for nagios plugins
+  file:
+    state: directory
+    dest: "{{ openvpn_nagios_plugin_dir }}"
+    mode: 0755
+    owner: root
+    group: root
+
+- name: Place openvpn check plugins
+  copy:
+    src: "{{ item }}"
+    dest: "{{ openvpn_nagios_plugin_dir }}/{{ item }}"
+    owner: root
+    group: root
+    mode: 0755
+  with_items:
+    - check_openvpn
+    - check_openvpn_clients.pl
+
+- name: Make status file readable for service checks
+  file:
+    path: "{{ openvpn_etcdir }}/{{ openvpn_status }}"
+    mode: 0644
+
+- name: Register consul service for openvpn
+  consul_service:
+    name: vpn-infra
+    port: "{{ openvpn_port }}"
+    checks:
+      - name: service check
+        type: script
+        script: service openvpn status || exit 2
+
+      - name: process check
```

Comment:
this might fail..b/c it checks the output of "ps -ef|grep ''openvpn" (in this case) and that returns all the commands related to openvpn and not just openvpn process...so if you are only tailing "openvpn.log" and no openvpn process is running, still it will pass the process check...so will need to pass openvpn process check regex in "--ereg-argument-array" to "script:" for proper openvpn process check...